### PR TITLE
Add telescope to "input" config section

### PR DESCRIPTION
**Pull request by jmeyers314**
Use the "input" config section to load the batoid telescope configuration, and then use the loaded telescope in both the batoid WCS and lsst_optics.  Placing the telescope input here ensures it is consistent in both places (and any future uses).

Note, this request is to merge into field-rotation; so after review, we should effect this merge first and then field-rotation -> main.

Also, for the moment, I removed the `shift_optics` functionality.  That'll get replaced very soon in a new PR to add perturbations to the telescope config input section.

### CODE REVIEW

File: imsim/photon_ops.py [hunk 1]
```diff
@@ -277,16 +272,15 @@ def get_camera_cached(camera_name: str):
 @photon_op_type("lsst_diffraction")
 def deserialize_lsst_diffraction(config, base, _logger):
     kwargs = config_kwargs(config, base, LsstDiffraction)
-    shift_optics = kwargs.pop("shift_optics", base.get("shift_optics", None))
 
     opsim_meta = galsim.config.GetInputObj(
         "opsim_meta_dict", config, base, "opsim_meta_dict"
     )
 
+    telescope = galsim.config.GetInputObj("telescope", config, base, "telescope")
```
Comment (rmjarvis) [inline imsim/photon_ops.py:280]: This should really be mentioned in the Register command that this input type is associated with this PhotonOp.  cf. http://galsim-developers.github.io/GalSim/_build/html/config_stamp.html#galsim.config.RegisterPhotonOpType

File: imsim/batoid_wcs.py [hunk 2]
```diff
@@ -572,8 +554,17 @@ def makeWCSFactory(self, boresight, rotTelPos, obstime, band, camera='LsstCam',
...
+
+    def quickFactory(self, boresight, rotTelPos, obstime, band, camera='LsstCam'):
+        from .telescope_loader import load_telescope
+        telescope = load_telescope(f"LSST_{band}.yaml", rotTelPos=rotTelPos)
+        return self.makeWCSFactory(
+            boresight, obstime, telescope, bandpass=band, camera=camera
+        )
 
 
 RegisterWCSType('Batoid', BatoidWCSBuilder())
```
Comment (rmjarvis) [inline imsim/batoid_wcs.py:570]: This should mention that it's associated with the Telescope input type now.
```
RegisterWCSType('Batoid', BatoidWCSBuilder(), input_type='telescope')
```
cf. http://galsim-developers.github.io/GalSim/_build/html/config_image.html#wcs-field

File: tests/test_batoid_wcs.py [hunk 3]
```diff
@@ -463,33 +462,37 @@ def test_config():
...
         temperature=temperature,
         pressure=pressure,
         H2O_pressure=H2O_pressure
     )
 
     config = {
+        'input': {
+            'telescope': {
+                'file_name':f"LSST_{band}.yaml",
+                'rotTelPos': {'type': 'Degrees', 'theta':rotTelPos.deg}
```
Comment (rmjarvis) [inline tests/test_batoid_wcs.py:483]: This is fine, but you could also just write
```
'rotTelPos': rotTelPos
```
In tests, you don't necessarily need to drop all the way back to a string-able yaml.  You can just give a complete object if you already have it.

**Review by rmjarvis — CHANGES_REQUESTED** [review]
LGTM.  Just a couple minor suggestions.

**Review by g-braeunlich — APPROVED** [review]
I like the idea of moving the telescope config to input.

**Comment by jmeyers314** [comment]
Thanks both.  @rmjarvis , I addressed your comments.  Let me know if you have any more.

**Review by rmjarvis — APPROVED** [review]

**Comment by rmjarvis** [comment]
LGTM

**Pull request merged**